lib/Extractor/KLEEBuilder.cpp

```diff
@@ -778,8 +791,19 @@ void ExprBuilder::setBlockPCMap(const BlockPCs &BPCs) {
   for (auto BPC : BPCs) {
     assert(BPC.B && "Block is NULL!");
     BlockPCPredMap &PCMap = BlockPCMap[BPC.B];
-    ref<Expr> PE = getInstMapping(BPC.PC);
     auto I = PCMap.find(BPC.PredIdx);
+    // Relying on a class-level flag may not be a nice solution,
+    // but it seems hard to differentiate two cases:
+    //   (1) UBInstExpr collected through blockpc, and;
+    //   (2) UBInstExpr collected through pc/lhs/rhs
+    // For the first case, UBInst(s) is conditional, i.e.,
+    // they are dependent on the fact that blockpc(s) are true.
+    if (I != PCMap.end()) {
+      UBInstHasPrecondition = true;
+      UBInstPrecondition = I->second;
+    }
+    ref<Expr> PE = getInstMapping(BPC.PC);
+    UBInstHasPrecondition = false;
```

Comment:
I don't see a better way of doing it now, but maybe @pcc has an idea. Otherwise, fine with me.